src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -81,6 +81,7 @@ internal void InitializeCurl()
                 SetCredentialsOptions(_handler.GetNetworkCredentials(_handler._serverCredentials,_requestMessage.RequestUri));
                 SetCookieOption(_requestMessage.RequestUri);
                 SetRequestHeaders();
+                SetSslOptions();
```

Comment:
We can go ahead and merge this, but considering that non-https and non-redirects are quite common, I would like us to look in a follow-up PR at whether there's any measurable impact to this, and if there is, if we can do this only when using https and only when redirecting from http to https.